uaclient/contract.py

```diff
@@ -93,8 +94,10 @@ def add_contract_machine(
                 code=response.code,
                 body=response.body,
             )
-
-        return response.json_dict
+        response_json = response.json_dict
+        if "machineToken" in response_json:
+            secret_manager.secrets.add_secret(response_json["machineToken"])
```

Comment:
Can't we use a `get` call here and just ignore the secret if it is empty on the secret manager method ?

Reply:
This also apply to all other similar checks in the code

Reply:
Thanks for that, also added a check when adding secrets so the `""` string is ignored if the `get` call returns an empty string in these cases.